Move Expression typings from Canvas to OSS Interpreter

lukeelmers opened the pull request:
### Summary
This is a first pass at moving the Canvas Expression Function types that @clintandrewhall worked on over to OSS so they can be exported from the interpreter for use by other plugins.

Reviewers: It's probably easiest to follow this by looking at each of the commits in order.

### Usage / Testing
```typescript
import { ExpressionFunction } from 'plugins/interpreter/types';
// ^^ I still need to verify that this webpack alias works in typescript from x-pack

const name = 'myfunc';

interface Context {
  type: 'whatever'; // in most cases you'll be importing Context from elsewhere
}

interface Arguments {
  foo: boolean | null;
}

interface Return {
  type: 'whatever'; // in most cases you'll be importing Return from elsewhere
}

export const myfunc = (): ExpressionFunction<typeof name, Context, Arguments, Return> => ({
  // function definition
});
```

### Notes / Thoughts
- Perhaps we should have an interpreter "types" base interface that individual types extend from when authoring their interfaces. Specifically we need to enforce that types have a `type` string value in their interface (which this generic relies on).
- I haven't copied over some of the other generics (`ContextFunction` and `NullContextFunction`) because this all seemed to work with one unified generic... but I may be missing edge cases so @clintandrewhall let me know if I'm missing anything!
- Next steps if this checks out will be: converting interpreter functions to use the new generic, making sure all types have interfaces defined, removing any types from canvas that this replaces
- It may make more sense to re-export from the `data` plugin now (where this will live long term), instead of interpreter

lukeelmers committed:
Move Expression types to OSS.

lukeelmers committed:
Add interfaces for KibanaContext and KibanaDatatable types.

lukeelmers committed:
Convert esaggs expression function to TypeScript.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/14890/)

elasticmachine commented:
Pinging @elastic/kibana-app-arch

elasticmachine commented:
Pinging @elastic/kibana-canvas

lukeelmers commented:
I'll mark as ready to review for now, and we can discuss whether (once this is approved) we should include all of the updates in Canvas in this PR, or a subsequent one.

chrisdavies commented on src/legacy/core_plugins/interpreter/common/types/kibana_datatable.ts:
```diff
@@ -19,30 +19,49 @@
+
+interface Column {
+  id: string;
+  name: string;
```
Does the column not contain any type information? That'd be pretty handy to have in a number of situations. (e.g. `type: 'string'`)

lukeelmers replied:
I don't think so -- kibana_datatable is just the tabified data format, e.g.

```javascript
{
  columns: [{ id: 'foo', name: 'Foo Col' }],
  rows: [{ foo: 'bar' }, { foo: 'baz' }],
}
```

monfera replied:
Chris, yes it'd be handy, it's not present at the moment, but typed table attribs had come up in discussion. For example, in the context of TS, it could warn that one wants to multiply a numeric column with an alpha column, or filter a time attribute represented as milliseconds since epoch with a range that represents time as days since epoch (ie. the physical type matches but the _attribute domains_ have mismatched semantics).

Also, semantic types would be great for something like.... helping determine Kibana Lens options :-) For example, if an editor knew that those numbers represent time in milliseconds, then appropriate time filters etc. can be added (example is made up, only to illustrate that physical, easy to infer types such as numbers don't help a downstream, uncoupled consumer very much). On the demo day prez for shareable dataviz I touched on using attribute types as a goal, and super early steps.

monfera replied:
@chrisdavies ^ I should've used the handle

chrisdavies commented on src/legacy/core_plugins/interpreter/common/types/kibana_datatable.ts:
```diff
@@ -19,30 +19,49 @@
+}
+
+interface Row {
+  [key: string]: object;
```
I don't think `object` is the right type, here. A row is generally key: string, value: primitive, no? Maybe this should be: `[key: string]: unknown` or maybe any instead of unknown?

lukeelmers replied:
oops, yes it is; thanks for catching that

chrisdavies commented on src/legacy/core_plugins/interpreter/types/common.ts:
```diff
@@ -0,0 +1,23 @@
+/**
+ * Utility type: extracts returned type from a Promise.
+ */
+export type UnwrapPromise<T> = T extends Promise<infer P> ? P : T;
```
My brain hurts!

lukeelmers replied:
Clint was already using this in a generic somewhere, so I just extracted it as a utility. Unfortunately I can't take credit for authorship xD

Use case is when you have a type that may or may not be returned from a Promise -- you can use this and then you don't need to worry about whether the promise is there or not.

chrisdavies replied:
Yeah, I understood it. It just took a bit of staring. I was just reminded of my experience learning Haskell a few years back. Hence, the aching brain. :)

chrisdavies reviewed (approved):
LGTM w/ possible tweaks. There are a few places where 'any' or 'object' is used, and 'unknown' might be more appropriate. Other than that, this looks great.

lukeelmers committed:
Clean up KibanaDatatable and KibanaContext types.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/15128/)

lukeelmers commented:
```
Caused by: org.elasticsearch.common.settings.SettingsException: The configuration setting [xpack.security.authc.realms.oidc.oidc1.rp.client_secret] is required
```

I think this is an unrelated issue that was fixed on master. Will rebase & push.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/15161/)

Commit:
[Code] refactor launcher code, add unit tests (#36863)

Commit:
feature(code/frontend): breadcrumb links to code root and project root (#34252)

Commit:
[Code] fix query string key conflict (#36620)

* fix code query string key conflict

* fix code go to definition url

XavierM committed:
[SIEM] Remove imports from server folder (#37037)

* remove import from server folder

* the other way around too

ppisljar committed:
moves ui out of index patterns service (#36348)

maryia-lapata committed:
[Tagcloud] Allow to send showLabel parameter with false value (#36563)

* Allow to send showLabel parameter with false value

* Add unit test

ppisljar commented:
i tried moving tagcloud fn over to TS, everthing seems to work, i am only missing types for Render type.

alexwizp committed:
[TSVB] Fix Rollup Search with auto interval for Rollup Jobs in calendar intervals (#36081)

* auto interval

* [TSVB] Rollup Search - 'auto' interval is not working for Rollup Jobs which were created in Calendar intervals

* [TSVB] Rollup Search - 'auto' interval is not working for Rollup Jobs which were created in Calendar intervals

* fix issue

* fix broken tests

* fix pr comments

* fix pr comments

vitalics committed:
[TSVB] Table tests (#36762)

* implement table tests

maryia-lapata committed:
Remove unused translations (#37069)

flash1293 committed:
Expressions service (#36885)

alexwizp committed:
[TSVB] Rollup Search - override index pattern functionality is not working with Rollup search (#36641)

Commit:
[Canvas] Adds telemetry info for custom elements (#36177)

* Adds telemetry info for custom elements

* Addresses PR feedback, converts to TS

* Fix bad equality check

* Fix problem with errored query

* Fix eslint issue with typescript

* Switch from export * to stop eslint complaining

* Fix test

* Clean up TS and tests

* Remove unused imports

chrisronline committed:
Once the buffer has any events, the collector is always ready (#36995)

chrisronline committed:
[Monitoring] Fix 500 error from /api/stats (#36986)

* Fix 500 and add test for it

* Return an empty object so the logic later will still execute

* Update tests

Commit:
[TSVB] Get rid of default export on TSVB (#36872)

The `import/no-default-export` has been turned on for TSVB.

* Get rid of default export on TSVB
* Change eslint rules lines

sorenlouv committed:
[APM] Fix encoding issue with forward slash in path (#37072)

stevedodson committed:
[ML] Improve anomalies in Kibana sample data (#36982)

* [ML] Sample ML jobs: Updates to make anomalies clearer

Minor updates to web logs data and ML job
definitions to make anomalous activity more
distinct.

* [ML] Sample ML jobs: Further updates to jobs and data

Adding missing files from previous commit and
changing job naming (removing weblog_* prefix etc.)

* [ML] Resolving PR review comments.

* [ML] Improve anomalies in Kibana sample data

Minor fixes:
- Rename job group to `kibana_sample_data` throughout
- Fix index pattern link in `sample_data_sets.ts`
- Fix log.json.gz count test

maryia-lapata committed:
[Vis] Move propFilter from ui/filter to ui/agg_types/filter (#36875)

* Move propFilter from ui/filter to ui/agg_types/filter

* Return default value

* Make propFilter generic

* Convert propFilter from a mocha test to a jest test

walterra committed:
[ML] Data Frames: Fix (error) messages, data field aggs, date picker check. (#36972)

- Layout and message fixes for displaying error messages for the pivot preview in `ml/public/data_frame/components/define_pivot/pivot_preview.tsx
- Fixes list of available date field aggregations.
- Fix to make sure datepicker is always disabled for pivot wizard.
- Fixes incomplete job delete confirm modal message.

igoristic committed:
Swapped the anchor tag with EuiLink (#37020)

Commit:
Revert "[Canvas] Adds telemetry info for custom elements (#36177)" (#37100)

This reverts commit 4fabe6c4410f98ab1776d8a81b033149537661cc.

igoristic committed:
Typecasting the attr boolean to the correct type (#37031)

Bargs committed:
Update QueryBarInput to accept index pattern strings (#36916)

For many use cases a consumer of the QueryBarInput (or QueryBar) might only have an index pattern string in hand. Instead of forcing every consumer to reimplement the fetching logic to get a full pattern object, this PR updates the QueryBarInput to do the fetching itself if the indexPatterns array prop contains any strings. If a string does not exactly match the title of any of the saved objects then we return the default index pattern instead.

clintandrewhall commented:
I'm patching this down to test as we speak... feedback shortly :-)

Commit:
[siem/ui] avoid importing server code in ui (#37103)

* [siem/ui] avoid importing server code in ui

* covert a couple extra imports

lukeelmers commented:
FYI, I'm learning there are some issues with `plugins/*` webpack alias imports from within TS files specifically. I think it's a tsconfig issue.

I'm looking into this now, but for the time being you may need to test with a relative import.

lukeelmers commented:
I'm also thinking I will move the types so they are re-exported from the top level of `plugins/interpreter`. This is a core convention that will be enforced in the new platform (you can only import from the top level of a plugin), so makes sense to do it here.

alvarezmelissa87 committed:
[ML] Anomaly explorer query bar - ensure anomalies table results retained after refresh (#37041)

* Don't specify influencers if overall cell selected

* use swimlane type overall constant

clintandrewhall commented:
Would it be possible to define these as a `.d.ts` file in `@kbn/interpreter`?  Canvas can't import the types as they are defined now, and I'd like future plugin developers to be able to use them easily.

Commit:
Split ciGroup3 and ciGroup5 (#36928)

* split large x-pack ciGroup5 into 3 groups

* split large x-pack ciGroup3 into 2 groups

* Add groups to the jenkins xpack ci group file

* Re-level some more work

* move es_search_source test to new describe block

* move es_search_source to first in ciGroup7

pmuellr committed:
update CONTRIBUTING.md re: linting with VSCode (#37005)

Adds a note in the Linting section about updating VSCode settings to
see "live" linting for TypeScript.

mattkime committed:
No unused vars cleanup set two (#37051)

* no unused vars cleanup

chrisronline committed:
Remove improper usage (#37099)

clintandrewhall commented on src/legacy/core_plugins/interpreter/types/arguments.ts:
```diff
@@ -0,0 +1,179 @@
+ * the `type` key as the string literal type for the argument.
+ */
+// prettier-ignore
+type KnownTypeToArgumentString<T> = 
```
We're using this for more than just arguments, (e.g. in `context` in the fn definition).  I'd rename this and put in common.

clintandrewhall commented on src/legacy/core_plugins/interpreter/types/arguments.ts:
```diff
@@ -0,0 +1,179 @@
+  T extends boolean ? 'boolean' : 
+  T extends number ? 'number' :
+  T extends null ? 'null' :
+  T extends { type: string } ? T['type'] :
```
We can't key off of `type` because sometimes the argument *name* is "type":

https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/canvas_plugin_src/functions/server/demodata/index.ts#L16

In addition, this blows up if the type contains a `type` property for a different purpose... consider `Filter`, which has a `type` which refers to the type of filter to be performed:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/canvas_plugin_src/functions/types/common.ts#L36

We may have to go with `__type`, unfortunately... which lends itself to a whole new set of challenges.

clintandrewhall reviewed (changes requested):
So far I've found a couple of problems.

lukeelmers commented:
Thanks for all your testing on this @clintandrewhall! Here are the remaining items left to sort out based on our offline conversation:

- We have one example (`Filter`) where type property is used for a different purpose. Will investigate whether this is something we can change in Canvas in the longer term so that all types are unified (perhaps we can even provide a base interface for types that enforces the presence of `type: string;`). In the meantime, will consider some short term solutions to deal with `Filter`.
- Will export types directly from `plugins/interpreter` instead of `plugins/interpreter/types` as this is something that will be enforced via linting rules in New Platform anyway. Will also look into a `kbnInterpreter` module as a possible route.
- I'll take a look at your testing branch so I can check all of this with Canvas locally.
- Still need to sort out the `tsconfig` issues with `plugins/*` imports.

chrisronline committed:
[Monitoring] Used fixed_interval explicitly for date_histogram aggregations (#37023)

* Used fixed interval explicitly

* Fix tests

toddself committed:
[feat] add reach-like functionality to object (#37118)

* [feat] add reach-like functionality to object

In #36804 we need to validate a `Partial<HttpConfig>` object which may
or may not have all the required keys attached to it.  This is the type
of use-case for `Joi.reach`, so we should expose a reach-like method on
the object validation class so that you can return the validator for a
specific key on an object.

* [fix] change to validateKey

* [fix] use throw error matcher

* [fix] use same interface for validate

* [fix] change test name]

Commit:
disable failing test

ycombinator committed:
[Grok Debugger] Trim trailing whitespace in sample data and grok patterns fields (#36245)

cqliu1 committed:
Remove unused propType (#37138)

cqliu1 committed:
Fixes check for existing min/max value in axisConfig (#37119)

spong committed:
[SIEM] Fix for removal of first filter clearing remaining filters (#37135)

## Summary

As outlined in https://github.com/elastic/ingest-dev/issues/419, clearing the first filter within the Timeline Query Builder would result in all the remaining `and`'d filters being cleared as well. This PR fixes that issues, and adds testing to catch a future regressions.

Before:
![filter_query_clear_bug](https://user-images.githubusercontent.com/2946766/58357586-a26f9200-7e38-11e9-8c5e-11bbdfcabfef.gif)


After:
![filter_clear_fix](https://user-images.githubusercontent.com/2946766/58357436-05145e00-7e38-11e9-8a51-b90481afb9b9.gif)


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) 
  - IE11 needs this fix to be applicable: https://github.com/elastic/ingest-dev/issues/263
- [ ] ~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- [ ] ~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] ~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

- [ ] ~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
- [ ] ~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

sorenlouv committed:
[APM] Add useMemo and minor feedback for environment filters (#36970)

* [APM] Add useMemo and minor feedback for environment filters

* Fix types

* Rename to `loadEnvironmentsFilter`

dgieselaar committed:
[APM] Make sure conditions under which transaction menu items are sho… (#37070)

* [APM] Make sure conditions under which transaction menu items are shown are correct

- check for url.domain for uptime link
- always show discover link
- show log link when traceId is set

* Add more tests for TransactionActionMenu

walterra committed:
[ML] Data Frames: Fix calendar_interval for date histograms. (#36905)

Fixes the use of calendar_interval instead of interval for date histograms.

dgieselaar committed:
[APM] Make sure URL hooks are in sync with history.location (#36676)

* [APM] Make sure URL hooks are in sync with history.location

Previously, the `useUrlParams` and `useLocation` hooks were possibly running out of sync with history.location, as they maintained their own state. By listening to history changes, they would re-render their context providers with the updated values. However, the context providers are wrapped by the `Router` component which re-renders the context providers when the location changes. This means that the initial render after a location change would have the updated values in `history.location`, but not in the values returned from the `useUrlParams` and `useLocation` hooks. These hooks would then (both) queue another render with the updated values being passed to the context.

This change lets the hooks piggyback on the re-renders from the `Router` component, and uses `history.location` to pass their derived values to the context. This ensures that the values returned from the hooks and `history.location` are always in sync. It should also remove a few unnecessary renders.

Fix tests

- Passes history as a prop to URLParamsProvider, easier to test and consistent with usage of LocationProvider
- Do not test behaviour of history.replace & re-renders, only whether history.replace was called with the right Location values. This is because re-rendering is no longer a concern of URLParamsProvider, we leave that up to the Router component now.

Address review feedback:

- Use history/location from Router so we have one source of truth for routing info
- For UrlParamsContext, avoid use of history.replace. The behaviour w/ history.replace is incorrect. We now use useRef and trick React into re-rendering with an "empty" state update.
- Unit test for whether useUrlParams does not queue an unnecessary render.

Memoize context value of UrlParamsProvider

* Move infra items back to the top again

Piggybacking on this change to fix an order issue in TransactionActionMenu that was introduced w/ 2e3fbd24a8fee57bb87eafd536fc3c6ed8c6d81b (woops).

walterra committed:
[ML] Data Frames: Fixes aggregation item overflow. (#37068)

Fixes an issue where long field names would overflow the aggregation item of the pivot config list. This fix was already done for the group-by-items, now brought over to aggregations too.

mw-ding committed:
[Code] Reset the default search scope (#37014)

mw-ding committed:
[Code] improve project status polling management (#36841)

* [Code] improve project status polling management

* [Code] Use make repository status polling cancellable

* change back to 1000ms polling interval

* deduplicate repository status polling runner before run

dgieselaar committed:
[APM] Navigate to default transaction sample (#36223)

* [APM] Navigate to default transaction sample

If the distribution data changes, and the currently selected transaction sample is no longer available, we navigate to the default sample of the latest distribution data.

Previously, a displayed sample would "stick" if the new distribution data did not have a default sample.

Readability improvements

Use location primitives

Simplifies hook logic in TransactionDetails/Distribution by only passing the necessary primitives (pathname, hash, search) to the history.replace() call rather than the full location object. Removes the need for memoization w/ useRef too.

Fix lint error

* Remove unused translations

* Simplify redirectToDefaultSample callback

* Use history.location instead of location prop

alexwizp committed:
TSVB:  selecting the colour picker crashes Kibana (#37176)

Fix: #37115

vitalics committed:
Tsfy screenshot service and comparepng internal lib (#37177)

vitalics committed:
[TSVB] markdown variables(revert fix) (#34067)

* add markdown tests

rashidkpc committed:
Fix Canvas fullscreen

Closes #34080

For whatever reason, `Style.it` is unwrapping and dropping the root node of the component it wraps. No idea why. This simply wraps the tree up in a sacrificial `<div/>` for `Style.it` to eat.

Commit:
Revert "Fix Canvas fullscreen"

This reverts commit f02c78b8b62a0941ffa980f0f5bc6b3c037b9c53

ogupte committed:
Closes #36992 by fixing path params bug by checking if serviceName even (#37062)

exists in the current pathname

kertal committed:
Add  automatic alignment to gauge visualisation (#34845)

Add flexibility to alignment of multiple gauge vis by supporting the following alignments: automatic, horizontal, vertical. the default is: automatic, it aligns the gauges depending on screen space to horizontal or vertical

Further Adaptions
- sample data
- migration of saved objects
- removal of jquery as depencency
- adaptions of rendering to prevent clipping

dmlemeshko committed:
TSVB: functional test for color picker, preview disable and series clone (#37186)

* [visualize/_tsvb_chart] test for color picker, changing  preview and cloning series

walterra committed:
[ML] Data Frames: Fix to avoid queries for indices with more than 1024 fields. (#37183)

- 7.0 introduced indices.query.bool.max_clause_count which defaults to 1024. This can break certain queries (e.g. simple_query) for indices which have more than 1024 fields (e.g. certain beats indices). The optional data frames query uses simple_query and could therefor break the source index preview as well the pivot preview and pivot job itself given these conditions.
- Originally the default query (* used for simple_query) was always applied for source index previews and pivot previews. A new check isDefaultQuery() will now allow a) the source index preview to use a more efficient match_all query and b) avoid adding the query to the pivot config. This avoids triggering the max_clause_count when no optional query is set.
- If an index has more than 1024 fields, the input form for an optional query will be hidden. A helper text explains the reasoning. This avoids triggering max_clause_count related errors from within the UI. A user can still copy a UI created config to the clipboard and add an optional query in Kibana dev console.
- Additionally, this PR adds a fix to format date fields in the source index preview table using moment-timezone and formatHumanReadableDateTimeSeconds to display dates with the correct timezone.

sulemanof committed:
Move all switches to the SwitchParamEditor (#37215)

kobelb committed:
[docs][skip ci] Infrastructure and Logs read only access (#35566)

* Adding a read only access section to the infrastructure docs

* Adjusting where the infrastructure read only access section is and
adding the logs readonly access section

* Fixing grammar

Commit:
Moved query filter and push filter into filter manager (#36764)

* Moved query filter and push filter from filter bar to filter manager

walterra committed:
[ML] Fix access denied for data frames. (#37178)

With insufficient privileges, a user would be redirected to the ML plugin's access-denied page which mentions the required user roles to access the pages. Since data frames introduces new user roles these messages were not correct.

This PR fixes it by redirecting to a specific access-denied page for data frames. To avoid to much refactoring as a fix, the page is a copy and port to React of the original one. In a follow up for 7.3, we should merge the two pages and it should have options to display required user roles given a certain context like anomaly detection or data frames.

lcawl committed:
[DOCS] Facilitates content re-use in the monitoring configuration (#37123)

walterra committed:
[ML] Fix to not add configs with names which would result in nesting conflicts (#37212)

Two group-by or aggregation names could result in conflicting nested fields which would return an error for pivot previews. For example, two names like responsetime and responsetime.avg are not allowed.

This PR fixes the issue by extending the tests whether a configuration is allowed to be added to the list of group-by and aggregation configurations. If a conflict is detected, a toast notification gets triggered and the configuration won't be added.

andrew-goldstein committed:
[SIEM] Fixes timeline notes overflowing the modal (#37134)

## [SIEM] Fixes timeline notes overflowing the modal (#37134)

Beginning in Chrome 74 (see elastic/eui#1902), the timeline notes modal can overflow in some browsers, per the _Before_ gif below.

This fix uses `EuiModal` and other `EUI` components to address the overflow issue across all the browsers tested in the gifs below:

### Before (Chrome `74.0`)

![01-before-chrome](https://user-images.githubusercontent.com/4459398/58355997-4d7d4d00-7e33-11e9-864b-7e77d0635116.gif)

### After (Chrome `74.0`)

![02-after-chrome-74 0](https://user-images.githubusercontent.com/4459398/58356028-6128b380-7e33-11e9-8c7d-3022e45b1f41.gif)

### After (Firefox `67.0`)

![03-after-firefox-67 0](https://user-images.githubusercontent.com/4459398/58356056-6ede3900-7e33-11e9-92ca-a6dd8e0b804b.gif)

### After (Safari `12.1.1`)

![04-after-safari-12 1 1](https://user-images.githubusercontent.com/4459398/58356123-a947d600-7e33-11e9-80ab-d6b3d3c601c0.gif)

### After (IE`11.0.9600`)

![05-after-ie-11](https://user-images.githubusercontent.com/4459398/58356131-aea52080-7e33-11e9-989e-51aab7c4e9da.gif)

https://github.com/elastic/ingest-dev/issues/442

alvarezmelissa87 committed:
return false for each jobId if it does not exist (#37116)

FrankHassanabad committed:
[SIEM] Low impact linter rules to start with (#37137)

## Summary

Low impact linter rules to help catch bugs for back-porting as well as moving into the future. We will slowly turn the warns into errors and fix the errors as we go along. This helps us avoid a "big bang" switch over approach to avoid back-port issues as we are porting bugs aggressively at the beginning.

Added a custom pattern detection rule for determining if the UI is trying to import anything from the server backend since that would cause web-pack front to pull in backend code. Also forbid the use of NodeJS imports on the front end.  

Double checked that I was not duplicating rules from:
* https://github.com/elastic/kibana/blob/master/packages/eslint-config-kibana/typescript.js
* https://github.com/elastic/kibana/blob/master/packages/eslint-config-kibana/javascript.js

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~
~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

XavierM committed:
fix persistence of timeline when you pinned or add a note to an event (#37248)

FrankHassanabad committed:
[SIEM] Fixed unescaped HTML entity issue (#37257)

## Summary
  * Turned on linter rule to be an error
  * Removed the extra character
  * https://github.com/elastic/ingest-dev/issues/467

Removes the extra `}` character seen after line 84 below in the screen shot:

<img width="368" alt="Screen Shot 2019-05-28 at 9 39 28 AM" src="https://user-images.githubusercontent.com/1151048/58498271-f4672f00-813a-11e9-8b80-2326db363bb0.png">

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

alvarezmelissa87 committed:
[ML] Prevent overflow of large metric values in data visualizer/file data visualizer cards (#36932)

* filedataviz cards - abbrev long numeric values

* add directive for index dataviz values

* Add arg to indicate format. update tests

* truncate with ellipsis

* use eui ellipsis truncation

* increase card content height

* ensure truncation for firefox and safari

Commit:
update yarn.lock file

FrankHassanabad committed:
[SIEM] Fix state issues to avoid potential React update issues (#37261)

## Summary

Fixed prevState issues with React setState
  * This will avoid subtle bugs with the react update renderer
  * Enabled one lint rule from warn to error
  * Turned off the other linter issue as it has an open ticket against it.
  * https://github.com/elastic/ingest-dev/issues/468

Summarize your PR. If it involves visual changes include a screenshot or gif.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

mshustov committed:
Expose plugin contracts to Legacy platform (#37218)

* expose plugin contracts to Legacy platform

* move mapToObject to src/core/utils

dmlemeshko committed:
[visualize/_shared_item] skip test due to #37130 (#37131)

Bargs committed:
Fix infinite loop in QueryBarInput's componentDidUpdate (#36946)

We were calling updateSuggestions unconditionally in componentDidUpdate
which in turn called setState unconditionally, so componentDidUpdate
was running in an infinite loop.

w33ble committed:
Fix: Fullscreen sizing on old workpads (#37289)

* fix: provide css value for workpads

old workpads were missing this value, this fixes scaling in fullscreen mode

* fix: make workpadCss a required prop

this will cause warnings in the future if the value is missing. also remove the unused css prop.

mw-ding committed:
[Code] Reenable lsp_indexer mocha tests and incooporate language filter test cases (#37266)

chongzhe committed:
[Code] change Typescript to TypeScript in UI (#37270)

jkakavas committed:
Remove validation from OIDC endpoint (#37159)

* Remove validation from OIDC endpoint

This change removes validation of query parameters from the
/api/security/v1/oidc endpoint. It was discovered during manual
testing that Google's OP is sending extra parameters than the ones
identified in https://tools.ietf.org/html/rfc6749#section-4.1.2
which is refernced by
https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse
(for instance auth_user and session_state). The existing validation
rules only allowed the expected query parameters but this
means that Kibana wouldn't be able to complete OpenID Connect
authentication with Google acting as the OP.
As dictated in the standard (RFC6749), "The client MUST ignore
unrecognized response parameters." so we should allow but discard
any extra parameters we do not recognize and not throw an error.

* address feedback

* address feedback

* add validation for the issuer

kertal committed:
Remove duplicate fields of rollup based index patterns (#37221)

Rollup fields were were merged twice with other fields, this caused duplicates. Fix of a bug introduced by #28762

angorayc committed:
[SIEM] Replace Eui chart with elastic charts for siem kpis (#36660)

* move charts to separate components

* replace areachart

* apply custom styles

* customize barchart color

* customize color for areachart

* move reusable functions into common

* exchange x & y value in barchart dataset

* replace pure component with react memo and upgrade enzyme adapter

alexwizp committed:
Visual Builder duration in second showing as number (#35813)

* Visual Builder duration in second showing as number

Fix: #35641 , #35805

* fix PR comments

* Visual Builder duration in second showing as number

* Visual Builder duration in second showing as number -fix placeholder

Commit:
Move filter bar and apply filters to data plugin (#36778)

* Setup filter plugin
With filter_bar and ApplyFiltersPopover
Update imports

* Updated i18n keys

* Typo fix

* update JP translation names

azasypkin committed:
Introduce Kerberos authentication provider. (#36112)

maryia-lapata committed:
[Vis Editor] Fix undefined error in Field control (#37179)

* Add default value for labels in sortByLabel function

* TS: make label optional

sulemanof committed:
[Vis: Default editor] EUIficate top_sort param editor (#36629)

* EUIficate top_sort param editor

* Set field if only one available

* Remove unused translations

* Fix functional tests

* Fix browser tests

* Rewrite functional test for combo_box field select

* Remove unused translations

* Update label

vitalics committed:
[TSVB] [AT] TimeSeries refactor (#36987)

* move tsvb timeseries tests into new file

sulemanof committed:
Upgrade enzyme and enzyme-adapter-react-16 to the latest (#37064)

* Upgrade enzyme and enzyme-adapter-react-16 to the latest

* Skip failed expectation

* Update yarn.lock

* Update snapshot

mshustov committed:
add docs how to use config in New Platform (#36782)

* add docs how to use config in NP

* do not encourage storing initializer context in plugin

maryia-lapata committed:
[Vis: Default editor] EUIficate controls of Geohash agg (#37213)

* EUIficate controls of geohash agg

* Fix tests

Commit:
[Code] fix lsp init unit test by using fake timers (#37201)

* [Code] fix LSP init unit test by using fake timers

mshustov committed:
introduce pre-/post-auth request hooks for HttpServer (#36690)

* introduce pre-,post-auth stages

* cleanup integration_tests. now contracts available in tests

* auth per route is configurable

* Unify lifecycle results structure

* expose api to store auth state and status via http service

* update tests

* update docs

* use full name, auth should not mutate request

* move basePath functionality under namespace

* regenerate docs

* Revert "move basePath functionality under namespace"

This reverts commit 9599d328013b0ba734294959b90454eeeb9382f3.

* Revert "regenerate docs"

This reverts commit 1799d3b0881b175ab2d36b312d1172806ab2a113.

* regenerate docs

* updated yarn.lock no idea why

* extract AuthStateStorage to a separate entity

* get rid of nested ifs

* describe what is the difference between hooks

* re-wording

sulemanof committed:
[Vis: Default editor] EUIficate top_aggregate and size param editors (#36567)

* EUIficate top_aggregate_and_size param editor

* Remove template

* Change typescript interfaces

* Fix browser tests

* Add an icon alert

* Changes due to comments

* Move error to a help text

* Move error to a field

* Change validation logic

* Fix discarding changes action

* Remove changed translation

maryia-lapata committed:
[Vis: Default editor] EUIficate Percentile Ranks and Percentiles (#35349)

* Migrate number-list

* Add types

* Refactoring

* Add number row

* Add and remove items

* Add onChange

* Add validate function

* Apply validation

* Fix merge conflict

* Apply order validation

* Fix ts

* Refactoring

* EUIficate percentiles control

* Remove unused translations

* Move helper functions to a separate file

* Fix code review comments

* Make add button 'xs' size

* Remove extra space

* Fix validation

* Remove unused translations

Bamieh committed:
[Docs] i18n docs add japanese (#37233)

sulemanof committed:
[Vis: Default editor] EUIficate order_agg param editor (#36984)

* EUIficate order_agg param editor

* Update browser tests

* Add types

* Update enzyme version in x-pack

* Fix functional tests

* Changes due to comments

* Update_terms_helper.tsx

Co-Authored-By: Maryia Lapata <mary.lopato@gmail.com>

* Fix code review comments

* Update yarn.lock

Bamieh committed:
[i18n] JP translations (#37229)

* updated JP translations

* run i18n check

legrego committed:
Security - remove auth scope provider (#36998)

* remove auth scope provider

* handle missing roles

* guard for unauthenticated calls

* update functional tests to not expect a scope property

* there's always money in the banana stand

* revert interceptor optimizations

* protect against missing roles

* address pr feedback

* remove scope as expected property on kerberos auth response

Commit:
[SIEM] Update Empty Page Messages (#37251)

* update EmptyPage comp and text on all pages

* change proptype from `string` to `IconType`

* update tests and snapshots

* remove translation imports

* restore `align-self` styles for IE11 centering

* proper apostrophe and string in single quotes, per frank

* update host and ip details

* DRYing up host and network empty page message

* fix localization

* fixing localization again

Commit:
[checks-reporter] reenable (#37280)

cqliu1 committed:
[Canvas] Fixes Workpad prop typo (#37351)

Conky5 committed:
Remove directory from dockerfiles target (#37258)

* Remove directory from dockerfiles target

Produce a docker build context tarball without a top-level directory to
match the expected format for downstream automation.

This will produce:

```
$ tar -tzf target/kibana-8.0.0-docker-build-context.tar.gz
Dockerfile
bin/
config/
bin/kibana-docker
config/kibana.yml
```

Instead of:

```
$ tar -tzf target/kibana-8.0.0-docker-build-context.tar.gz
kibana-8.0.0-docker-build-context/Dockerfile
kibana-8.0.0-docker-build-context/bin/
kibana-8.0.0-docker-build-context/config/
kibana-8.0.0-docker-build-context/bin/kibana-docker
kibana-8.0.0-docker-build-context/config/kibana.yml
```

* Use options object for root directory config

Commit:
[Feature/Reporting] Export Saved Search CSV as Dashboard Panel Action (#34571)

XavierM committed:
[SIEM] fix data provider query when the field is a timestamp (#37281)

* fix dataprovider query when the filed is a timestamp

* review II

* review III

timroes committed:
Add addError function to toastNotifications (#32187)

mattkime committed:
bump github-checks-reporter to delay 60s on retries and retry 10x (#37358)

toddself committed:
[feat] create additional http servers (#36804)

* [feat] create additional http servers

allow for additional http servers to be created, tracked and returned

* respond to pr feedback

* tweak test

* update documentation

* destructure port, remove unnecessary imports

* [fix] export correct type

* [feat] expose createNewServer to plugins

* [fix] respond to pr feedback

* todo: add schema validation & integration test

* use reach

* [fix] use validateKey to validate partial

* [fix] change config shadowing

* check kibana port & prevent shadowing

* centralize start/stop for servers, add integration test

* remove unnecessary property

* never forget your await

* remove option to pass config into start

* fix pr feedback

* fix documentation

* fix test failures

renovate[bot] committed:
Configure Renovate (#37385)

* Add renovate.json

* Update and rename renovate.json to renovate.json5

* Update renovate.json5

* Update renovate.json5

* extend the base config

* remove enabled:false

* skip the part limiting managers, we point to specific package.json files instead

* try to disable updating all but eslint packages

* nest npm specific config

* fix syntax error

* bad copy job

* tweak labels

* disable update requests for `@kbn/` packages

* stop overriding rangeStategy

* extend package.json selectors

* add a couple more package.json files

zfy0701 committed:
[Code] change suggested clone url (#37354)

w33ble committed:
Fix: handle null argument values (#37411)

typeof null === 'object', so add an explicit check for null

clintandrewhall committed:
Fix formatdate not allowing strings (#37301)

* Fix formatdate not allowing strings

* Addressing Feedback

zfy0701 committed:
[Code] add util scripts to run tests/checks (#37365)

lukeelmers committed:
Update ArgumentString-related types to simply be TypeString.

lukeelmers committed:
Rename types.ts to index.ts so types can be imported at the top level.

lukeelmers committed:
Resolve merge conflicts.

lukeelmers commented:
ugh. sorry for the ping everyone 👋  

you can safely ignore this

lukeelmers commented:
To spare y'all from further updates, I'm going to go ahead and close this one in favor of #37438.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/16000/)
